drop-transfer/src/file/id.rs

```diff
@@ -92,6 +92,15 @@ impl FileSubPath {
     pub fn name(&self) -> &str {
         self.0.last().expect("Missing last path component")
     }
+
+    pub fn from_path(path: impl AsRef<Path>) -> crate::Result<Self> {
+        let vec = path
+            .as_ref()
+            .iter()
+            .map(|cmp| cmp.to_str().map(String::from).ok_or(crate::Error::BadPath))
+            .collect::<Result<_, _>>()?;
+        Ok(Self(vec))
+    }
```

Comment:
Could you provide more context on what this does and what was wrong with the previous approach?

Reply:
This one constructs the `FileSubPath` from path components. Previously we only had a method that constructed it by splitting the path string on `/` which is clearly incorrect for Windows.
I mean both methods are needed because the string representation used in the WS protocol is always using `/` but when constructing from the actual filesystem path we need to use this method

Reply:
Thanks for explaining, make sense